### ISSUE TITLE
[HOLD for payment 2022-11-22] [$500] mWeb - Request Money- Missing cursor when requesting/split the money

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#10693]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Issue found when executing #10147 
## Action Performed:
1. Go to staging.new.expensify.com
2. Log in with any account
3. Click FAB > Request Money
4. Enter an amount
5. Move cursor to the middle of the amount
 5.1. Enter a Digit
5.2. Delete a digit
5.3. Paste a couple of digits
5.4. Cut a couple of digits

## Expected Result:
Cursor should be present when requesting/split the money

## Actual Result:
Missing cursor when requesting/split the money

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Mobile Web (Android)

**Version Number:** 1.1.94.4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/187492774-2cd40f6e-7598-42eb-8fc5-b004ebea5cba.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** Applause internal team
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by adelekennedy** [comment]
[internal](https://www.upwork.com/ab/applicants/1564889519573094400/job-details)
[external](https://www.upwork.com/jobs/~0151b41ed5756729f1)

**Comment by AndrewGable** [comment]
This is a regression from https://github.com/Expensify/App/issues/6154 - I believe @eVoloshchak should fix this.

**Comment by trjExpensify** [comment]
I'm also a bit confused by the steps in the OP versus the video. Something isn't clear, because I can see the cursor in the video? 😕 

I also can't seem to reproduce it, if that's the problem as stated. 


https://user-images.githubusercontent.com/16232057/187634101-d693cf6d-c642-41b7-9c18-f9cf7ceefe83.MP4

**Comment by trjExpensify** [comment]
Oh, I see. @kavimuru can you confirm that, please?

**Comment by kbecciv** [comment]
@trjExpensify Correct, mWeb - Andriod

**Comment by trjExpensify** [comment]
Cool, thanks for clarifying.

**Comment by vinicius-grandi** [comment]
# Proposal

## Problem
The textbox is losing focus when number pad is pressed.

## Solution
You should pass the method focusTextInput to the number pad and use it when onPress event occurs

**_src/pages/iou/steps/IOUAmountPage.js_**
```diff
...
<BigNumberPad
    numberPressed={this.updateAmountNumberPad}
    longPressHandlerStateChanged={state => this.shouldUpdateSelection = !state}
+   focusTextInput={this.focusTextInput}
/>
...
```
_**src/components/BigNumberPad.js**_
```diff
...
<Button
    ...
+   onPress={() => {
+       this.props.focusTextInput()
        this.props.numberPressed(column);
+   }}
    ...
/>
...
```

**Comment by vinicius-grandi** [comment]
> > The textbox is losing focus when number pad is pressed
> 
> @vinicius-grandi can we make it not loose the focus in the first place?
# Proposal

Just use preventDefault on pressOut event (which would normally cause focus on button). The side effect is that prevents the click event as well, so to solve this, fire click event manually.

**_src/components/BigNumberPad.js_**
```diff
<Button 
...
    onPressOut={(e) => {
+     e.preventDefault();
+     e.target.click();
      clearInterval(this.state.timer);
      ControlSelection.unblock();
      this.props.longPressHandlerStateChanged(false);
    }}
...
/>
...
```

**Comment by adelekennedy** [comment]
@thienlnam thoughts on the proposal above?

**Comment by thienlnam** [comment]
Sorry, just got back from OOO

@rushatgabhane Did you have a chance to test those changes locally? The proposal looks good but want to make sure it doesn't impact other button places this is used

**Comment by thienlnam** [comment]
There seem to be a few cursor issues that are related - the fix should encapsulate these issues

https://github.com/Expensify/App/issues/11095

**Comment by thienlnam** [comment]
Pending proposal review in https://github.com/Expensify/App/issues/11095

**Comment by trjExpensify** [comment]
@thienlnam, should this issue be on hold for that one then?

**Comment by parasharrajat** [comment]
Reviewed the other issue https://github.com/Expensify/App/issues/11095. I think may be another regression from https://github.com/Expensify/App/pull/10147 or a missed test case during the review. 

But the reprodcution of this issue is not covered in https://github.com/Expensify/App/issues/11095.

**Comment by thienlnam** [comment]
I know we've had some cursor issues trying to remove setNativeProps to upgrade to fabric https://github.com/Expensify/App/issues/11053.  This might be part of that as well but the issue you linked also could be the cause. In that case though, we can open this back up for proposals

**Comment by thienlnam** [comment]
Going to be OOO so re-assigning this so we can move forward with proposals
Also cc @adelekennedy this would be a good candidate for double!

**Comment by adelekennedy** [comment]
@neil-marcellini Jack is ooo - reassigned this issue to you!

**Comment by adelekennedy** [comment]
Per the last convo we're opening this back up, I'll double this once we hit the 7 day mark!
[Internal](https://www.upwork.com/ab/applicants/1584574150975074304/job-details)
[External](https://www.upwork.com/jobs/~014440ed5cbc8e4cb2)

**Comment by neil-marcellini** [comment]
Still waiting for proposals.

**Comment by adelekennedy** [comment]
Doubling time!

**Comment by thienlnam** [comment]
Proposal looks fine but since some linked issues have been closed, 

@kavimuru can we confirm this is still happening?

**Comment by kbecciv** [comment]
Issue is reproduced with build 1.2.23.9

https://user-images.githubusercontent.com/93399543/200013280-868d9077-b15a-4e6b-8db2-baa4b23bb66f.mp4

**Comment by thienlnam** [comment]
Thanks! 🟢 to the proposal

**Comment by adelekennedy** [comment]
my first checklist!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by thienlnam** [comment]
I added a link to the potential offending PR but it's from way back - https://github.com/Expensify/App/pull/10147

cc @adelekennedy Are you able to take care of the other items on the checklist please? https://github.com/Expensify/App/issues/10693#issuecomment-1311038367

**Comment by adelekennedy** [comment]
Hired!

**Comment by thienlnam** [comment]
Yeah, this PR should still be eligible for the 50% bonus - my bad should have clarified in this issue.  Going to reopen so that can get taken care of

**Comment by thienlnam** [comment]
cc @adelekennedy for adding the bonus payment

**Comment by adelekennedy** [comment]
Bonuses paid out!
